instrumentation/spring/spring-web/spring-web-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springweb/v3_1/SpringWebInstrumentationModule.java

```diff
@@ -29,6 +30,15 @@ public ElementMatcher.Junction<ClassLoader> classLoaderMatcher() {
         .and(not(hasClassesNamed("org.springframework.web.ErrorResponse")));
   }
 
+  @Override
+  public void registerHelperResources(HelperResourceBuilder helperResourceBuilder) {
+    // make the filter class loadable by ClassPathResource - in some cases (e.g. spring-guice)
+    // Spring might want to read the class file metadata; this line will make the filter class file
+    // visible to the bean class loader
+    helperResourceBuilder.registerForAllClassLoaders(
```

Comment:
Wouldn't it suffice to use `register` instead of `registerForAllClassLoaders`? 
WDYT about adding a link to the gh issue in the comment?
Perhaps rephrase the comment a bit, `ClassPathResource` doesn't really load the class but rather loads the content of the class file.

Reply:
> Wouldn't it suffice to use `register` instead of `registerForAllClassLoaders`? 

Great point -- I had to remember why I used `registerForAllClassLoaders` in the actuator instrumentation in the first place 😅 

> WDYT about adding a link to the gh issue in the comment? Perhaps rephrase the comment a bit, `ClassPathResource` doesn't really load the class but rather loads the content of the class file.

👍